tools/rule_extractor/rule_extractor.py

```diff
@@ -0,0 +1,155 @@
+
+        endpoint = "https://{}:{}/api/prometheus/api/v1/{}".format(
+            self.ip, self.port, route
+        )
+        try:
+            response = requests.get(url=endpoint, verify=False)
+        except requests.exceptions.ConnectionError as exc:
+            raise Exception("Failed to reach endpoint {}: {!s}".format(
+                route, exc
+                )
+            )
+        if response.status_code != 200:
+            raise Exception("Expected status code <200> but got {}".format(
+                response.status_code
+                )
+            )
+
+        return response.json()
+
+    def write_to_file(self, content, rule_name):
+        with open('{}.json'.format(rule_name), 'w', encoding='utf-8') as f:
+            f.write(json.dumps(content, ensure_ascii=True))
+
+    def get_dict_element(self, data, path, delimiter='.'):
+        return functools.reduce(operator.getitem, path.split(delimiter), data)
+
+
+def main():
+    """
+        python rule_extractor --ip 127.0.0.1 --port 9090 --type rules
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-i',
+        '--ip',
+        dest='ip',
+        help='Specify the target control-plane ingress IP address'
+    )
+    parser.add_argument(
+        '-p',
+        '--port',
+        dest='port',
+        default='8443',
+        help='Specify the target control-plane ingress port'
+    )
+    parser.add_argument(
+        '-t',
+        '--type',
+        required=True,
+        dest='rule_type',
+        help='Specify the rule type. For example: rules or alerts'
+    )
+
+    args = parser.parse_args()
+    if not args.ip:
+        # Obtain VM ip by connecting to it and running a command
+        # Requires that the VM is up and running.
+        ip = subprocess.check_output(
+            [
+                "vagrant",
+                "ssh",
+                "--command",
+                "hostname -I | cut -d' ' -f1"
+            ]
+        ).decode(sys.stdout.encoding).strip()
+        # Validate that we got a real ipv4 address
+        try:
+            ipaddress.ip_address(ip)
+        except ValueError:
+            sys.stderr.write("Ip address: {} is invalid".format(ip))
+        except Exception as exc:
+            sys.stderr.write(
+                "Syntax error while checking ip address validity: {!s}".format(
+                    exc
+                )
+            )
+        alertrule = AlertRuleFile(ip=ip, port=args.port)
+    else:
+        alertrule = AlertRuleFile(ip=args.ip, port=args.port)
+
+    if args.rule_type == "rules":
+        # Returns list of rules that are currently loaded
+        out = alertrule.query_prometheus_api(route="rules")
+        alertrule.write_to_file(out, args.rule_type)
+        # For documentation purposes, we don't want to litter the whole thing
+        # with all fields, so lets strip off some "ambiguous" fields
+        rule_group = alertrule.get_dict_element(out, "data.groups")
```

Comment:
Wouw, that's a bit over engineered, no ?
```suggestion
        rule_group = alertrule.get('data', {}).get('groups', [])
```